x-pack/legacy/plugins/infra/server/kibana.index.ts

```diff
@@ -7,17 +7,15 @@
 import { i18n } from '@kbn/i18n';
 import { Server } from 'hapi';
 import JoiNamespace from 'joi';
+import { PluginSetupContract as UsageCollection } from 'src/plugins/usage_collection/server';
 import { initInfraServer } from './infra_server';
 import { compose } from './lib/compose/kibana';
 import { UsageCollector } from './usage/usage_collector';
 import { inventoryViewSavedObjectType } from '../common/saved_objects/inventory_view';
 import { metricsExplorerViewSavedObjectType } from '../common/saved_objects/metrics_explorer_view';
 
-export interface KbnServer extends Server {
-  usage: any;
-}
-
-export const initServerWithKibana = (kbnServer: KbnServer) => {
+export const initServerWithKibana = (kbnServer: Server) => {
+  const usageCollection = kbnServer.newPlatform.setup.plugins.usageCollection as UsageCollection;
```

Comment:
So ideally in the NP and if we were shimmed correctly here, this would all happen in the `setup` method right?

Reply:
@jasonrhodes correct. you can check the (dev notes) for a full example.